### PR TITLE
remover: make idempotent

### PR DESCRIPTION
**Pull request by ryantm**
Why
===
* If we cannot find the dep, it either already has been removed or we have no hope of removing it. Either way, we can return the unchanged contents instead of an error.
* This gives us idempotency on the removing operation, which is nice to have.
* nix-editor already has idempotent semantics on adding

What changed
===
* Add unit test for remover
* If we can't find the dep return the original contents unchanged

Test plan
===
* cargo test

### CODE REVIEW

File: src/remover.rs [hunk 1]
```diff
@@ -8,8 +8,11 @@ pub fn remove_dep(
 ) -> Result<String> {
     let remove_dep = remove_dep_opt.context("error: expected dep to remove")?;
 
-    let range_to_remove = find_remove_dep(deps_list, &remove_dep)
-        .context("error: could not find dependency to remove")?;
+    let search = find_remove_dep(deps_list, &remove_dep);
+    if search.is_err() {
+        return Ok(contents.to_string());
```
Comment (cdmistman) [inline src/remover.rs:13]: can we change the function to return `Result<Option<String>>` and returning `None` here instead? that way we can avoid updating the mtime of the dirent and won't trigger nix rebuilds :)

**Review by cdmistman — APPROVED** [review]
otherwise lgtm

File: src/remover.rs (hunk 1 above)
Comment (ryantm) [reply src/remover.rs:13]: We [don't write the file here](https://github.com/replit/nix-editor/blob/671ea01ad18c149f46330426a229c4f6c616467a/src/main.rs#L289) if the contents are the same.

Comment (ryantm) [reply src/remover.rs:13]: I confirmed the mtime is not updating myself. I agree that you're way will be better, will add it next time I dive into the remover code.

**Pull request merged**